pkg/webui/console/containers/device-onboarding-form/device-type-form-section/device-type-repository-form-section/index.js

```diff
@@ -38,6 +49,30 @@ import BandSelect from './device-selection/band-select'
 
 import style from './repository.styl'
 
+const encodeBrand = (value, oldValue) =>
+  value !== oldValue.brand_id
+    ? {
+        brand_id: value,
+        model_id: undefined,
+        hardware_version: undefined,
+        firmware_version: undefined,
+        band_id: undefined,
+      }
+    : oldValue
+const encodeModel = (value, oldValue) =>
+  value !== oldValue.model_id
+    ? {
+        brand_id: oldValue.brand_id,
+        model_id: value,
+        hardware_version: undefined,
+        firmware_version: undefined,
+        band_id: undefined,
+      }
+    : oldValue
+
+const decodeBrand = value => (isPlainObject(value) ? value.brand_id : '')
+const decodeModel = value => (isPlainObject(value) ? value.model_id : '')
```

Comment:
I used the decoder/encoder pattern for the `brand_id` and `model_id` fields because using the `onChange`-handlers would cause the form values to be updated with faulty values for a short time until the `setValues()` hook was executed. This would not be a big issue usually but in this case, it was since the added `useEffect()` hook on line 131 would fire whenever a completed set of version IDs was detected. If a version ID that contained a faulty combination of values was there, it would fire request for these values which would fail.